### PR TITLE
Add a forced by-node shards deallocation to all the allocation strategies

### PR DESCRIPTION
**Pull request by sergiy0**

### CODE REVIEW

File: src/main/scala/com/evolutiongaming/cluster/ExtendedShardAllocationStrategy.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      case _          => false
+    }
+    override def hashCode(): Int = address.hashCode
+  }
+
+  @volatile
+  private var nodesToForcedDeallocation: Set[Node] = Set.empty
+
+  def deallocateShardsFromNode(regionGlobalAddress: Address): Unit =
+    nodesToForcedDeallocation += new Node(regionGlobalAddress)
```
Comment (t3hnar) [inline src/main/scala/com/evolutiongaming/cluster/ExtendedShardAllocationStrategy.scala]: what if node has been restarted ?

Comment (t3hnar) [inline src/main/scala/com/evolutiongaming/cluster/ExtendedShardAllocationStrategy.scala]: what cluster keep running without downtime, however we are continually restarting nodes ?

File: src/main/scala/com/evolutiongaming/cluster/ExtendedShardAllocationStrategy.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  system: ActorSystem,
+  implicit val ec: ExecutionContext,
+  maxSimultaneousRebalance: Int,
+  deallocationTimeout: FiniteDuration) extends ShardAllocationStrategy {
+
+  val addressHelper = AddressHelperExtension(system)
+  import addressHelper._
+
+  class Node(val address: Address, added: Instant = Instant.now()) {
+    def expired: Boolean = Instant.now() isAfter (added plusMillis deallocationTimeout.toMillis)
```
Comment (t3hnar) [inline src/main/scala/com/evolutiongaming/cluster/ExtendedShardAllocationStrategy.scala]: are there any benefits from listening cluster events approach VS this approach

Comment (sergiy0) [inline src/main/scala/com/evolutiongaming/cluster/ExtendedShardAllocationStrategy.scala]: listening cluster events - is a good idea

File: src/main/scala/com/evolutiongaming/cluster/ExtendedShardAllocationStrategy.scala (hunk 1 above)
Comment (sergiy0) [inline src/main/scala/com/evolutiongaming/cluster/ExtendedShardAllocationStrategy.scala]: this is for "deploy w/o downtime" scenario

**Pull request merged**